### ISSUE TITLE
Documentation referring to old psc-ide

### ISSUE THREAD

**Issue by hakonrossebo** [nwolverson/vscode-ide-purescript#87]
Hi,

I'm working on the requirements for an Elm language server (https://github.com/hakonrossebo/elm-language-server-requirements-specification) and while trying to describe the the components used by vscode-ide-purescript I see that there are some references to https://github.com/kRITZCREEK/psc-ide. (Also mentioned here https://github.com/nwolverson/purescript-ide-purescript-core). Is this still in use, or is this only old documentation?

I've written a short summary of how vscode-ide-purescript works in the bottom of this document https://docs.google.com/document/d/1ETeZeN17hqM4yui4iqBv1jwO8HryDZyel2kdcxTCGGA/edit?usp=sharing if you want to correct me, or add any information (the document is open for editing and comments).

If you have any opinions on implementing the language server and language server protocol, I would be happy to know :)

**Comment by nwolverson** [comment]
You're right, that document is out of date, "psc-ide" was merged into the main compiler binary (now `purs ide`), the link you have https://github.com/purescript/purescript/tree/master/psc-ide would now be correct. (The purescript-ide-purescript-core project is not really important, just shared code between the vscode and atom plugins which I maintain)

The general architecture is that there is an "IDE server" as part of the compiler providing general services, but I have wrapped that in a separate node wrapper which is a language server (LSP). This is largely for legacy reasons, that an IDE server with its own protocol and editor integrations already exists, much of the wrapper code (written in PureScript) already existed, and it was easy to use the existing node libraries for writing a language server. Starting from scratch we'd probably take a more direct approach.

(Incidentally my future plan is to extract the language server into a separate repository & integrate with the atom language server support / ide UI that was recently announced).
